src/systems/interactions.js

```diff
@@ -0,0 +1,205 @@
+        hovered: null,
+        held: null,
+        spawning: null
+      }
+    };
+
+    this.cursorController = document.querySelector("#cursor-controller");
+  },
+
+  getRightRemoteHoverTarget() {
+    return this.rightRemoteHoverTarget;
+  },
+
+  tickInteractor(options, state) {
+    const userinput = AFRAME.scenes[0].systems.userinput;
+    if (state.held) {
+      const networked = state.held.components["networked"];
+      const lostOwnership = networked && networked.data && networked.data.owner !== NAF.clientId;
+      if (userinput.get(options.dropPath) || lostOwnership) {
+        state.held = null;
+      }
+    } else {
+      state.hovered = options.hoverFn.call(this, options.entity.body);
+      if (state.hovered) {
+        if (userinput.get(options.grabPath)) {
+          const offersConstraint =
+            state.hovered.components.tags && state.hovered.components.tags.data[options.constraintTag];
+          const isHoldableButton = state.hovered.components.tags && state.hovered.components.tags.data.holdableButton;
+          const superSpawner = state.hovered.components["super-spawner"];
+          if (offersConstraint || isHoldableButton) {
+            if (
+              this.el.is("frozen") ||
+              !state.hovered.components.pinnable ||
+              !state.hovered.components.pinnable.data.pinned
+            ) {
+              state.held = state.hovered;
+            }
+          } else if (superSpawner) {
+            this.spawnObjectRoutine(state, options, superSpawner);
+          }
+        }
+      }
+    }
+  },
+
+  tick2() {
+    if (!this.el.is("entered")) return;
+    this.rightHandTeleporter =
+      this.rightHandTeleporter || document.querySelector("#player-right-controller").components["teleporter"];
+
+    this.tickInteractor(this.options.leftHand, this.state.leftHand);
+    if (!this.state.rightRemote.held) {
+      this.tickInteractor(this.options.rightHand, this.state.rightHand);
+    }
+
+    if (!this.state.rightHand.held && !this.state.rightHand.hovered) {
+      this.tickInteractor(this.options.rightRemote, this.state.rightRemote);
+    }
+
+    const enableRightRemote =
+      !this.state.rightHand.hovered && !this.state.rightHand.held && !this.rightHandTeleporter.isTeleporting;
+    this.cursorController.components["cursor-controller"].enabled = enableRightRemote;
+    if (!enableRightRemote) {
+      this.state.rightRemote.hovered = null;
+    }
+
+    if (this.el.systems.userinput.get(paths.actions.logInteractionState)) {
+      console.log(
+        "Interaction System State\nleftHand held",
+        this.state.leftHand.held,
+        "\nleftHand hovered",
+        this.state.leftHand.hovered,
+        "\nrightHand held",
+        this.state.rightHand.held,
+        "\nrightHand hovered",
+        this.state.rightHand.hovered,
+        "\nrightRemote held",
+        this.state.rightRemote.held,
+        "\nrightRemote hovered",
+        this.state.rightRemote.hovered
+      );
```

Comment:
I don't want 
```
console.log(this.state);
```
because I want to make sure I see a _snapshot_ of the values, rather than getting a "live" reference that will only become a snapshot once clicked on in the console.